### PR TITLE
🔍 Added zoom controls, search bar, and legend to map

### LINKED ISSUE #22: Create map UI overlay elements
**Summary**
Make UI overlay elements for map:
- Search bar
- Legend (be able to open and close)
- Zoom controls

**Target (check Figma for latest designs)**
image

**Acceptance Criteria / User Stories**
- As a user, I should be able to 
  - zoom in and out with the zoom controls
  - click and type in search bar (no results)
  - open and close legend (default open)

### PR DESCRIPTION
**Pull request by abetco**
### Changes
Added zoom controls, search bar, and legend to map.
Did some integration with navbar and sidebar container.

Notes:
- From the design, it seems like we might have multiple legends. Shouldn't be too hard to offer that flexibility. The way the legend is displayed right now is through an img, and unless we want be really flexible, we can just do images for all the different legends. Might want a higher res image though, mine was from a screenshot of the design.
- Legend icon is an image as well, could maybe do an svg as well, unless we want to use something standard for icons?
- Legend also doesn't match design currently in that opened legend doesn't match exactly, to offer that might need a decent amount more work. The current way I'm doing the open and closing is through  \<summary\> and \<details\> tag.
- Searchbar has no functionality
- Also since I added new public asset images, might need to redeploy? Locally shows the images fine but preview doesn't.

Fixes #22 

### Screenshots (if UI change)

image

image

image

image

image

---
### Checklist
- [ ] Added a11y fields (if applicable)

### CODE REVIEW

File: client/src/components/Map.jsx [hunk 1]
```diff
@@ -48,7 +50,51 @@ const Map = () => {
       mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_API_KEY}
       dragRotate={false}
       touchRotate={false}
-    />
+    >
+      <div
+        className="searchBar"
+        style={{ position: "absolute", left: 20, top: 20 }}
```
Comment (rxun) [inline client/src/components/Map.jsx:58]: not totally sure, but it is possible that you would want to put `role='search'` in this div, then for the aria-labels for input and button inside, you would remove redundant "search" words --> ie) `aria-label="Submit search" `would turn into `aria-label="Submit"`
reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Search_role

File: client/src/components/Map.jsx [hunk 2]
```diff
@@ -48,7 +50,51 @@ const Map = () => {
...
+          <summary class="p-2">
+            <img
+              class="w-5 h-5 inline mr-1 mb-1"
+              src="information.png"
+              alt="i"
+            />
+            Legend
+          </summary>
+          <div class="w-64">
+            <img src="legend.png" alt="legend" />
```
Comment (rxun) [inline client/src/components/Map.jsx]: probably something more descriptive is useful here -- for example, we would want to say `alt="Individual Arrests 0 to 30+"`

**Review by kellydunleavy — APPROVED** [review]
this looks great!

**Review by rxun — CHANGES_REQUESTED** [review]
i think what's there looks good, but check with siraj for final confirmation on css. i would be careful because right now on the preview, if you double click the icon / legend, it'll zoom in, so i would try and prevent that behavior. if you check the figmas as well, i believe siraj has added another set of toggles in the bottom left (i'd ask for updates from him) which should be fairly quick. i added some notes about a11y but looks good overall

**Review by rxun — APPROVED** [review]

**Pull request merged**